builds/e2e/templates/e2e-setup.yaml

```diff
@@ -63,6 +65,34 @@ steps:
     downloadType: single
     artifactName: $(artifactName)
 
+- pwsh: |
+    $github_headers = @{'Accept' = 'application/vnd.github.v3+json'; 'Authorization' = 'token $(TestGitHubAccessToken)'}
+
+    # Pagination doesn't seem to work for this call. It will become slower as the
+    # number of runs increases.
+    $actions_runs = Invoke-WebRequest -Headers $github_headers `
+        "https://api.github.com/repos/azure/iot-identity-service/actions/runs?event=push&branch=main" | `
+    ConvertFrom-JSON
+
+    cd $(Build.SourcesDirectory)
+    $aziot_commit = git submodule | awk '{print $1;}'
+    $artifacts_link = $actions_runs.workflow_runs | `
+        where {($_.head_sha -eq $aziot_commit) -and ($_.name -eq 'packages')} | `
+        Select-Object -ExpandProperty artifacts_url
+
+    $artifacts = Invoke-WebRequest -Headers $github_headers `
+        $artifacts_link | ConvertFrom-JSON
+    $download_link = $artifacts.artifacts | where {$_.name -eq '$(identityServiceArtifactName)'} | `
+        Select-Object -ExpandProperty archive_download_url
+
+    Invoke-WebRequest -Headers $github_headers $download_link -OutFile iot-identity-service.zip
+    Expand-Archive -Path iot-identity-service.zip -DestinationPath iot-identity-service
+
+    $packages = Get-ChildItem -Recurse iot-identity-service -Filter $(identityServicePackageFilter)
+    $packagePath = Convert-Path '$(System.ArtifactsDirectory)/$(artifactName)'
+    Copy-Item $packages -Destination $packagePath
```

Comment:
This is fine for now. We should discuss long term plans after this PR. This assumes that the aziot commit id is always coming from mainline (which isn't always true if we decided to do preview versions). Possible solutions - 
1) Stand up a new pipeline for iot-identity-service (which is redundant with Github Actions).; Maybe we switch everything to Github Actions?
2) Allow passing in branch and commit id as parameters into the pipeline. If not specified, we can do what you just did here.

WDYT?

Reply:
Option 2 makes sense IMO. We'll probably need to discuss further with other test automation folks.